lib/kernel.coffee

```diff
@@ -11,20 +11,18 @@ WatchSidebar = require './watch-sidebar'
 
 module.exports =
 class Kernel
-    constructor: (@kernelSpec, @grammar, @config, @configPath, @onlyConnect = false) ->
+    constructor: (@onDestroy, @kernelName, @kernelSpec, filePath, @config, @configPath, @onlyConnect = false) ->
         console.log 'Kernel spec:', @kernelSpec
         console.log 'Kernel configuration:', @config
         console.log 'Kernel configuration file path:', @configPath
-        @kernelName = @kernelSpec.display_name
+        @displayName = @kernelSpec.display_name
```

Comment:
Rather than rename it, I would remove it.
No need to duplicate the info already provided in the kernel spec.